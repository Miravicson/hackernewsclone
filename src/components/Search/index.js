import React from "react"
import PropTypes from "prop-types"

const Search = ({ searchTerm, onChange, children, onSubmit, searchRef }) => (
  <form onSubmit={onSubmit}>
    <input
      type="text"
      name="search"
      id=""
      onChange={onChange}
      value={searchTerm}
      ref={searchRef}
    />
    <button type="submit">{children}</button>
  </form>
)

export default Search

Search.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  searchRef: PropTypes.func
}
