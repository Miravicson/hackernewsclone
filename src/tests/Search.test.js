import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"

import Search from "../components/Search"

describe("Search", () => {
  const dummyProps = {
    searchTerm: "redux",
    onChange: () => {},
    onSubmit: () => {}
  }
  it("renders without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(<Search {...dummyProps}>Search</Search>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  test("it should match snapshot", () => {
    const component = renderer.create(<Search {...dummyProps}>Search</Search>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
