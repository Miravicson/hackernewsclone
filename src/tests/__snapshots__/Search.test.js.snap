// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search it should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id=""
    name="search"
    onChange={[Function]}
    type="text"
    value="redux"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
